# Tugas Besar I IF2200 Probabilitas dan Statistika
### Penarikan Kesimpulan dan Pengujian Hipotesis

Dibuat oleh: <br>
Dhanika Novlisariyanti (13521132) <br>
Brigita Tri Carolina (13521156) <br>

## Load Data

In [3]:
import numpy as np 
import pandas as pd 
import matplotlib as plt

df = pd.read_csv("anggur.csv")
df

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0             5.90            0.4451       0.1813        2.049401   0.070574   
1             8.40            0.5768       0.2099        3.109590   0.101681   
2             7.54            0.5918       0.3248        3.673744   0.072416   
3             5.39            0.4201       0.3131        3.371815   0.072755   
4             6.51            0.5675       0.1940        4.404723   0.066379   
..             ...               ...          ...             ...        ...   
995           7.96            0.6046       0.2662        1.592048   0.057555   
996           8.48            0.4080       0.2227        0.681955   0.051627   
997           6.11            0.4841       0.3720        2.377267   0.042806   
998           7.76            0.3590       0.3208        4.294486   0.098276   
999           5.87            0.5214       0.1883        2.179490   0.052923   

     free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0              16.593818                 42.27   0.9982  3.27       0.71   
1              22.555519                 16.01   0.9960  3.35       0.57   
2               9.316866                 35.52   0.9990  3.31       0.64   
3              18.212300                 41.97   0.9945  3.34       0.55   
4               9.360591                 46.27   0.9925  3.27       0.45   
..                   ...                   ...      ...   ...        ...   
995            14.892445                 44.61   0.9975  3.35       0.54   
996            23.548965                 25.83   0.9972  3.41       0.46   
997            21.624585                 48.75   0.9928  3.23       0.55   
998            12.746186                 44.53   0.9952  3.30       0.66   
999            16.203864                 24.37   0.9983  3.29       0.70   

     alcohol  quality  
0       8.64        7  
1      10.03        8  
2       9.23        8  
3      14.07        9  
4      11.49        8  
..       ...      ...  
995    10.41        8  
996     9.91        8  
997     9.94        7  
998     9.76        8  
999    10.17        7  

[1000 rows x 12 columns]

Mengecek apakah ada data yang null

In [4]:
df.isnull().sum()

fixed acidity           0
volatile acidity        0
citric acid             0
residual sugar          0
chlorides               0
free sulfur dioxide     0
total sulfur dioxide    0
density                 0
pH                      0
sulphates               0
alcohol                 0
quality                 0
dtype: int64

## 1. Deskripsi Statistika
Menulis deskripsi statistika (Descriptive Statistics) dari semua kolom pada data yang bersifat numerik, terdiri dari mean, median, modus, standar deviasi, variansi, range, nilai minimum, maksimum, kuartil, IQR, skewness dan kurtosis. Boleh juga ditambahkan deskripsi lain.

In [5]:
df.describe()

fixed acidity  volatile acidity  citric acid  residual sugar  \
count    1000.000000       1000.000000  1000.000000     1000.000000   
mean        7.152530          0.520839     0.270517        2.567104   
std         1.201598          0.095848     0.049098        0.987915   
min         3.320000          0.139900     0.116700        0.032555   
25%         6.377500          0.456100     0.237800        1.896330   
50%         7.150000          0.524850     0.272200        2.519430   
75%         8.000000          0.585375     0.302325        3.220873   
max        11.490000          0.805100     0.409600        5.550755   

         chlorides  free sulfur dioxide  total sulfur dioxide      density  \
count  1000.000000          1000.000000           1000.000000  1000.000000   
mean      0.081195            14.907679             40.290150     0.995925   
std       0.020111             4.888100              9.965767     0.002020   
min       0.015122             0.194679              3.150000     0.988800   
25%       0.066574            11.426717             33.785000     0.994600   
50%       0.082167            14.860346             40.190000     0.996000   
75%       0.095312            18.313098             47.022500     0.997200   
max       0.140758            27.462525             69.960000     1.002600   

                pH    sulphates      alcohol      quality  
count  1000.000000  1000.000000  1000.000000  1000.000000  
mean      3.303610     0.598390    10.592280     7.958000  
std       0.104875     0.100819     1.510706     0.902802  
min       2.970000     0.290000     6.030000     5.000000  
25%       3.230000     0.530000     9.560000     7.000000  
50%       3.300000     0.595000    10.610000     8.000000  
75%       3.370000     0.670000    11.622500     9.000000  
max       3.710000     0.960000    15.020000    10.000000

In [12]:
def describeStatistics(df):
    print("Mean\t: ", df.mean())
    print("Median\t: ", df.median())
    print("Modus\t: ", df.mode().iat[0])
    print("Standar Deviasi\t: ", df.std())
    print("Variansi\t: ", df.var())
    print("Range\t\t: ", df.max() - df.min())
    print("Nilai Minimum\t: ", df.min())
    print("Nilai Maksimum\t: ", df.max())
    print("Q1\t\t: ", df.quantile(0.25))
    print("Q2\t\t: ", df.quantile(0.50))
    print("Q3\t\t: ", df.quantile(0.75))
    print("IQR\t\t: ", df.quantile(0.75) - df.quantile(0.25))
    print("Skewness\t: ", df.skew())
    print("Kurtosis\t: ", df.kurtosis())
    
    

### Deskripsi Statistika Fixed Acidity

In [13]:
describeStatistics(df["fixed acidity"])

Mean	:  7.15253
Median	:  7.15
Modus	:  6.54
Standar Deviasi	:  1.2015975764938258
Variansi	:  1.4438367358358357
Range		:  8.17
Nilai Minimum	:  3.32
Nilai Maksimum	:  11.49
Q1		:  6.3774999999999995
Q2		:  7.15
Q3		:  8.0
IQR		:  1.6225000000000005
Skewness	:  -0.028878575532660055
Kurtosis	:  -0.019292120932933532


### Deskripsi Statistika Volatile Acidity

In [14]:
describeStatistics(df["volatile acidity"])

Mean	:  0.5208385000000001
Median	:  0.52485
Modus	:  0.5546
Standar Deviasi	:  0.09584827405534951
Variansi	:  0.009186891639389388
Range		:  0.6652
Nilai Minimum	:  0.1399
Nilai Maksimum	:  0.8051
Q1		:  0.4561
Q2		:  0.52485
Q3		:  0.585375
IQR		:  0.12927499999999997
Skewness	:  -0.1976986986092083
Kurtosis	:  0.16185290336961788


### Deskripsi Statistika Citric Acid

In [15]:
describeStatistics(df["citric acid"])

Mean	:  0.27051699999999995
Median	:  0.2722
Modus	:  0.3019
Standar Deviasi	:  0.04909837147076348
Variansi	:  0.0024106500810810814
Range		:  0.29290000000000005
Nilai Minimum	:  0.1167
Nilai Maksimum	:  0.4096
Q1		:  0.2378
Q2		:  0.2722
Q3		:  0.302325
IQR		:  0.064525
Skewness	:  -0.045576058685017296
Kurtosis	:  -0.1046792495951605


### Deskripsi Statistika Residual Sugar

In [16]:
describeStatistics(df["residual sugar"])

Mean	:  2.5671036825067595
Median	:  2.519430272865794
Modus	:  0.032554525015195
Standar Deviasi	:  0.9879154365046929
Variansi	:  0.9759769096842579
Range		:  5.5182004097078625
Nilai Minimum	:  0.032554525015195
Nilai Maksimum	:  5.550754934723058
Q1		:  1.896329943488683
Q2		:  2.519430272865794
Q3		:  3.220873482829786
IQR		:  1.3245435393411031
Skewness	:  0.13263808618992312
Kurtosis	:  -0.04298003436476261


### Deskripsi Statistika Chlorides

In [17]:
describeStatistics(df["chlorides"])

Mean	:  0.08119515250784977
Median	:  0.0821669021645236
Modus	:  0.0151224391657095
Standar Deviasi	:  0.020110647243996742
Variansi	:  0.00040443813257247374
Range		:  0.1256351302653488
Nilai Minimum	:  0.0151224391657095
Nilai Maksimum	:  0.1407575694310583
Q1		:  0.06657363190977357
Q2		:  0.0821669021645236
Q3		:  0.09531150148556258
IQR		:  0.028737869575789013
Skewness	:  -0.05131929742072573
Kurtosis	:  -0.2465081359240382


### Deskripsi Statistika Free Sulfur Oxide

In [18]:
describeStatistics(df["free sulfur dioxide"])

Mean	:  14.907679251029796
Median	:  14.860346236568924
Modus	:  0.194678523326937
Standar Deviasi	:  4.888099705756562
Variansi	:  23.893518733417388
Range		:  27.26784690109891
Nilai Minimum	:  0.194678523326937
Nilai Maksimum	:  27.462525424425845
Q1		:  11.426716949457617
Q2		:  14.860346236568924
Q3		:  18.313097915395005
IQR		:  6.886380965937388
Skewness	:  0.007130415991143398
Kurtosis	:  -0.36496364342685306


### Deskripsi Statistika Total Sulfur Oxide

In [19]:
describeStatistics(df["total sulfur dioxide"])

Mean	:  40.290150000000004
Median	:  40.19
Modus	:  35.2
Standar Deviasi	:  9.965767376218295
Variansi	:  99.3165193968969
Range		:  66.80999999999999
Nilai Minimum	:  3.15
Nilai Maksimum	:  69.96
Q1		:  33.785
Q2		:  40.19
Q3		:  47.0225
IQR		:  13.237500000000004
Skewness	:  -0.024060026812269975
Kurtosis	:  0.06394978916172311


### Deskripsi Statistika Fixed Density

In [20]:
describeStatistics(df["density"])

Mean	:  0.9959253000000001
Median	:  0.996
Modus	:  0.9959
Standar Deviasi	:  0.0020201809426487133
Variansi	:  4.081131041041045e-06
Range		:  0.013799999999999923
Nilai Minimum	:  0.9888
Nilai Maksimum	:  1.0026
Q1		:  0.9946
Q2		:  0.996
Q3		:  0.9972
IQR		:  0.0025999999999999357
Skewness	:  -0.07688278915513917
Kurtosis	:  0.01636562128503849


### Deskripsi Statistika pH

In [21]:
describeStatistics(df["pH"])

Mean	:  3.30361
Median	:  3.3
Modus	:  3.34
Standar Deviasi	:  0.10487548220040166
Variansi	:  0.010998866766766766
Range		:  0.7399999999999998
Nilai Minimum	:  2.97
Nilai Maksimum	:  3.71
Q1		:  3.23
Q2		:  3.3
Q3		:  3.37
IQR		:  0.14000000000000012
Skewness	:  0.14767259510827038
Kurtosis	:  0.0809095518741838


### Deskripsi Statistika Sulphates

In [22]:
describeStatistics(df["sulphates"])

Mean	:  0.5983899999999999
Median	:  0.595
Modus	:  0.59
Standar Deviasi	:  0.10081900799141187
Variansi	:  0.010164472372372372
Range		:  0.6699999999999999
Nilai Minimum	:  0.29
Nilai Maksimum	:  0.96
Q1		:  0.53
Q2		:  0.595
Q3		:  0.67
IQR		:  0.14
Skewness	:  0.1491989008699043
Kurtosis	:  0.06481928180859686


### Deskripsi Statistika Alcohol

In [23]:
describeStatistics(df["alcohol"])

Mean	:  10.592279999999999
Median	:  10.61
Modus	:  9.86
Standar Deviasi	:  1.5107060052287586
Variansi	:  2.282232634234234
Range		:  8.989999999999998
Nilai Minimum	:  6.03
Nilai Maksimum	:  15.02
Q1		:  9.56
Q2		:  10.61
Q3		:  11.622499999999999
IQR		:  2.0624999999999982
Skewness	:  -0.01899140432111647
Kurtosis	:  -0.13173155932281988


### Deskripsi Statistika Quality

In [24]:
describeStatistics(df["quality"])

Mean	:  7.958
Median	:  8.0
Modus	:  8
Standar Deviasi	:  0.9028017783827471
Variansi	:  0.8150510510510509
Range		:  5
Nilai Minimum	:  5
Nilai Maksimum	:  10
Q1		:  7.0
Q2		:  8.0
Q3		:  9.0
IQR		:  2.0
Skewness	:  -0.08905409122491781
Kurtosis	:  0.10829100232871003


## 2. Visualisasi
Membuat Visualisasi plot distribusi, dalam bentuk histogram dan boxplot untuk setiap kolom numerik. Berikan uraian penjelasan kondisi setiap kolom berdasarkan kedua plot tersebut.


### Visualisasi Fixed Acidity

### Visualisasi Volatile Acidity

### Visualisasi Citric Acid

### Visualisasi Residual Sugar

### Visualisasi Chlorides

### Visualisasi Free Sulfur Oxide

### Visualisasi Total Sulfur Dioxide

### Visualisasi Density

### Visualisasi pH

### Visualisasi Sulphates

### Visualisasi Alcohol

### Visualisasi Quality

## 3. Penentuan Apakah Tiap Kolom Berdistribusi Normal atau Tidak
Menentukan setiap kolom numerik berdistribusi normal atau tidak. Gunakan normality test yang dikaitkan dengan histogram plot.

### Fixed Acidity

### Volatile Acidity

### Citric Acid

### Residual Sugar

### Chlorides

### Free Sulfur Dioxide

### Density

### pH

### Sulphates

### Alchohol

### Quality

## 4. Tes Hipotesis Satu Sampel
Melakukan test hipotesis 1 sampel, <br>
a. Nilai rata-rata pH di atas 3.29? <br>
b. Nilai rata-rata Residual Sugar tidak sama dengan 2.50? <br>
c. Nilai rata-rata 150 baris pertama kolom sulphates bukan 0.65? <br>
d. Nilai rata-rata total sulfur dioxide di bawah 35? <br>
e. Proporsi nilai total Sulfat Dioxide yang lebih dari 40, adalah tidak sama dengan 50% ? <br>

## 5. Tes Hipotesis Dua Sampel
a. Data kolom fixed acidity dibagi 2 sama rata: bagian awal dan bagian akhir kolom. Benarkah rata-rata kedua bagian tersebut sama? <br>
b. Data kolom chlorides dibagi 2 sama rata: bagian awal dan bagian akhir kolom. Benarkah rata-rata bagian awal lebih besar daripada bagian akhir sebesar 0.001? <br>
c. Benarkah rata-rata sampel 25 baris pertama kolom Volatile Acidity sama dengan rata-rata 25 baris pertama kolom Sulphates? <br>
d. Bagian awal kolom residual sugar memiliki variansi yang sama dengan bagian akhirnya? <br>
e. Proporsi nilai setengah bagian awal alcohol yang lebih dari 7, adalah lebih besar daripada, proporsi nilai yang sama di setengah bagian akhir alcohol? <br>